lib/bolt/plugin/puppet_connect_data.rb

```diff
@@ -29,6 +49,13 @@ def validate_resolve_reference(opts)
           raise Bolt::ValidationError,
                 "puppet_connect_data plugin requires that 'key' be specified"
         end
+        if @input_data_path && !@data.key?(opts['key'])
```

Comment:
I think we should use a more explicit variable (or have a wrapper method) to indicate that what we've turned on is behavior to fail on unspecified keys.

That would make the initialize method like

```rb
if ENV.key?(INPUT_DATA_VAR)
  @fail_on_missing_key = true
  data_path = ENV[INPUT_DATA_VAR]
else
  @fail_on_missing_key = false
  data_path = File.join(context.boltdir, 'puppet_connect_data.yaml')
end
```

Reply:
I thought about that. The reason I didn't choose it was b/c that variable doesn't explain why we're failing on missing keys, while `input_data_path` does (because when it's input data, you want to fail there). That is also a personal preference so if you think the more explicit variable would help I can do that.

Think I'll still need to store the input data path to properly generate the error message.

Reply:
I think making `data_path` an instance variable is fine so we store the path in either case. I would be okay leaving this line as-is with a comment explaining. I worry that it will become unclear in the future why the behavior is different depending on whether you have an explicit path set or not.

Reply:
Updated with a comment.